app/test_main.py

```diff
@@ -1 +1,29 @@
-# write your code here
+import pytest
+
+from unittest import mock
+
+from app.main import can_access_google_page
+
+
+@pytest.mark.parametrize(
+    "url, internet_connection, access_status",
+    [
+        (False, False, "Not accessible"),
+        (True, False, "Not accessible"),
+        (False, True, "Not accessible"),
+        (True, True, "Accessible")
+    ]
+)
+@mock.patch("app.main.valid_google_url")
+@mock.patch("app.main.has_internet_connection")
+def test_can_access_page(
+        mock_valid_url: any,
+        mock_has_internet: any,
```

Comment:
You should not use `any` in the type annotation for mock, use the following to be more specific:
```suggestion
        mock_valid_url: mock.Mock,
        mock_has_internet: mock.Mock,
```